### ISSUE TITLE
"ignore_reciever_permission" and "ignore_sender_permission" parts for formatting

### ISSUE THREAD

**Issue by Squeej** [kangarko/ChatControl#467]
Would make managing formats alot simpler if it was possible to ignore a permission:) So if they have the permission, they're denied sending or receiving that part of the format

**Comment by ElBananaa** [comment]
> Would make managing formats alot simpler if it was possible to ignore a permission:) So if they have the permission, they're denied sending or receiving that part of the format

Hey!
Happy New Year!

Please follow the suggestion format. It's there for a reason.
You can actually do that by simply not giving the permission, or by denying it in the format.

**Comment by Squeej** [comment]
I did do it as a suggestion haha, for some reason it doesnt seem to have sent that way.

But yeah not giving the permission seems to default to everyone. Is there a deny part that can be used? Didn't see it in the wiki

**Comment by ElBananaa** [comment]
> I did do it as a suggestion haha, for some reason it doesnt seem to have sent that way.
> 
> But yeah not giving the permission seems to default to everyone. Is there a deny part that can be used? Didn't see it in the wiki

Actually there's no way to deny the permission in the plugin. You must deny it/set it to false in your permission plugin to deny it.

**Comment by Squeej** [comment]
Yeah I tried that, but I use group manager so it only kind of works. removing the permission in a group doesn't seem to do the trick, I have to actually give the negative permission in an individual command

I did notice they kinda broke since the latest update though, not even general permissions are being picked up by chat controls formats, but I havent done enough testing to determine if maybe it's just something on my end

EDIT: ah yeah so I tested it a bit and "receiver_permission " does seem to be kinda finicky, only works with certain parts of the format. It appears to not work when sender permission is paired with it, I'll update if I can confirm that is the case, version 10.1.14

**Comment by broken1arrow** [comment]
> Yeah I tried that, but I use group manager so it only kind of works. removing the permission in a group doesn't seem to do the trick, I have to actually give the negative permission in an individual command
> 
> I did notice they kinda broke since the latest update though, not even general permissions are being picked up by chat controls formats, but I havent done enough testing to determine if maybe it's just something on my end
> 
> EDIT: ah yeah so I tested it a bit and "receiver_permission " does seem to be kinda finicky, only works with certain parts of the format. It appears to not work when sender permission is paired with it, I'll update if I can confirm that is the case, version 10.1.14


I has done this mess and it only a part of it.  sender is perms when you send and Receiver is for you will se the msg or not.

```
  Third_Part:
    Message: '&e {message} '
    Sender_Permission: chatcontrol.helpmessage.player
    Receiver_Permission: chatcontrol.hjelp
  fjarde_Part:
    Message: '&e {message} '
    Sender_Permission: chatcontrol.helpmessage.staff
    Receiver_Permission: chatcontrol.helpmessage.player.se_staff_msg
```

**Comment by ElBananaa** [comment]
> Yeah I tried that, but I use group manager so it only kind of works. removing the permission in a group doesn't seem to do the trick, I have to actually give the negative permission in an individual command
> 
> I did notice they kinda broke since the latest update though, not even general permissions are being picked up by chat controls formats, but I havent done enough testing to determine if maybe it's just something on my end
> 
> EDIT: ah yeah so I tested it a bit and "receiver_permission " does seem to be kinda finicky, only works with certain parts of the format. It appears to not work when sender permission is paired with it, I'll update if I can confirm that is the case, version 10.1.14

Hey!

I'm currently using parts with receiver and sender permissions on my own server, and it works perfectly fine, so I guess this is something on your end.
```
Parts:
  delete-chat:
    Message: '&7[#cf0e00&lX&7]&r '
    Receiver_Permission: chatcontrol.removemessages
    Hover:
    - '&cRemove this message from the globalchat.'
    Run_Command: /{label} internal remove {message_uuid}

  report-player:
    Message: '&c⚠&r '
    Sender_Permission: chatcontrol.allowreport
    Receiver_Permission: chatcontrol.allowreport
    Hover:
    - '&e&lClick to report &c{player}&e.'
    Suggest_Command: /report create {player}
```

**Comment by Squeej** [comment]
I was able to solve it for the most part by downsizing the format, it was pretty complicated so I wouldn't be surprised at all if it's something on my end. Here's an example of what the global chat used to look like when I originally posted this suggestion. My original plan was to have the server name prefix for everyone, 3 prefixes only mods can see when regular players send a message, 3 prefixes only mods can see when they send messages without seeing the previous 3, seperate brackets for regular players, seperate brackets for mods without seeing other brackets, and seperate brackets for players that only mods can see without seeing other brackets.

haha, well anyway I'll just leave this here, if anyone wants to look through it and see what I did wrong I'd love to hear it. "chatcontrol.send" is a permission I made up on the spot:

https://pastebin.com/raw/ZEbqQjdU

Edit: another suggestion for the devs (I don't want to keep making tickets): An afk feature
